catalog/core/catalog-core-localstorageprovider/src/main/java/org/codice/ddf/catalog/content/impl/StorageProviderImpl.java

```diff
@@ -235,110 +240,128 @@ public DeleteStorageResponse delete(DeleteStorageRequest deleteRequest)
+                String[] parts = getContentFilePathParts(contentIdDir.getFileName()
+                        .toString());
+                Path target = Paths.get(baseContentDirectory.toAbsolutePath()
+                        .toString(), parts[0], parts[1], parts[2]);
+                try {
+                    if (Files.exists(target)) {
+                        List<Path> files = listPaths(target);
                         for (Path file : files) {
                             Files.deleteIfExists(file);
                         }
-                        Files.deleteIfExists(contentIdDir);
-                        Path part1 = contentIdDir.getParent();
-                        List<Path> part1Files = listPaths(part1);
-                        if (part1Files.size() == 0) {
-                            Files.deleteIfExists(part1);
-                        }
-                        Path part0 = part1.getParent();
-                        List<Path> part0Files = listPaths(part0);
-                        if (part0Files.size() == 0) {
-                            Files.deleteIfExists(part0);
+                    }
+                    if (Files.deleteIfExists(target)) {
+                        LOGGER.debug("Remove existing content id directory for commit: " + id);
+                    }
+                    Files.createDirectories(target.getParent());
+                    Files.move(contentIdDir, target, StandardCopyOption.REPLACE_EXISTING);
+                } catch (IOException e) {
+                    LOGGER.warn(
+                            "Unable to move files by simple rename, resorting to copy. This will impact performance.",
+                            e);
+                    try {
+                        if (Files.deleteIfExists(target)) {
+                            LOGGER.debug("Remove existing content id directory for commit: " + id);
                         }
-                    } catch (IOException e) {
-                        throw new StorageException("Could not delete file: " + metacard.getId());
+                        Path createdTarget = Files.createDirectories(target);
+                        List<Path> files = listPaths(contentIdDir);
+                        Files.copy(files.get(0), Paths.get(createdTarget.toAbsolutePath()
+                                .toString(), files.get(0)
+                                .getFileName()
+                                .toString()));
+                    } catch (IOException e1) {
+                        throw new StorageException("Unable to commit changes for request: " + id,
+                                e1);
                     }
                 }
-            } finally {
-                rollback(id);
             }
+        } finally {
+            rollback(request);
         }
     }
 
     @Override
-    public void rollback(String id) throws StorageException {
+    public synchronized void rollback(StorageRequest request) throws StorageException {
```

Comment:
This method looks a lot more like a clean up method than a rollback. Shouldn't a rollback delete the files that were created from the store, restore the previous version of the files that were updated and restore the files that were deleted?